### ISSUE TITLE
How to add auxiliary loss function into the model and how will it be trained?

### ISSUE THREAD

**Issue by yuejunpeng** [open-mmlab/mmsegmentation#1455]
I want to insert a module into psp-net framework, which need an auxiliary loss function to help train the model.
In the module what I want to add, the new loss has its own way to do back-propogation, which involves different variables and different way.
So will the framework automatically populate the loss correctly and do the back-propogation in the way what I want?
What should I do to train the auxiliary loss function?
It would be appreciated if you could answer the question!
Thank you very much!

**Comment by yuejunpeng** [comment]
I just don't know whether the computational graph of the auxiliary loss will be computed correctly, and whether the auxiliary loss will be stored into "parameters" class and back-propogate in its way.

**Comment by RockeyCoss** [comment]
Could you provide a more detailed description? Like, does the module computes its loss when its `forward` function is called?

**Comment by yuejunpeng** [comment]
OK. The module we want to insert use the dataset label to help train the model, so it has a auxiliary loss to train. Besides, we have a main loss function to mainly train the output. The auxilary loss just play an auxiliary role in the model. Now we can get the auxiliary loss (know how to populate it), but we don't know how to train the auxiliary loss function.

**Comment by RockeyCoss** [comment]
Well, I think the easiest way to achieve this is to write the module as a DecodeHead. And use it as an auxiliary head.

**Comment by yuejunpeng** [comment]
Thank you very much! We'll try it at once.

**Comment by jaideep11061982** [comment]
@yuejunpeng  were you able to achieve it ?

**Comment by jaideep11061982** [comment]
if yes could u paste the details

**Comment by jaideep11061982** [comment]
if yes could u paste the details
